### PR TITLE
feat: more sections and configurable diffing

### PR DESCRIPTION
**Pull request by Julusian**
Depends on https://github.com/nrkno/tv-automation-atem-connection/pull/111

Builds on https://github.com/nrkno/tv-automation-atem-state/pull/18

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

Feature

* **What is the new behavior (if this is a feature change)?**

New sections of state

* Colour generators
* Advanced Chroma Keyer
* Multiviewers
* Classic audio
* Fairlight audio

* **Other information**:

Not tested yet

### CODE REVIEW

**Review by baltedewit — COMMENTED** [review]
Good additions overall. My biggest concern is that some users of the library (particularly those using it through https://github.com/nrkno/tv-automation-state-timeline-resolver) may not want to override the multiviewers through the state library.

**Comment by Julusian** [comment]
I've made a start on prototyping selective state diffing.

The diffState function takes an object describing what should be diffed.
The basic design of the object is:
* Everything is optional to make building a minimal object easy
* For every 'array' in the state, it should be possible to select all, none, or just some
* There is a helper to make it create the object to allow for diffing of everything
* It should be granular, as adding more levels later will be hard to do without it being a breaking change

File: src/diff.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+		multiviewer: {
+			properties: true,
+			windows: 'all',
+		},
+	}
+}
+
+export interface SectionsToDiff {
+	auxiliaries?: DiffAuxiliaries
+	multiviewer?: DiffMultiViewer | DiffMultiViewer[]
```
Comment (baltedewit) [inline src/diff.ts:14]: Is there an 'all' missing here?

**Comment by baltedewit** [comment]
I think I agree with this approach in general. I assume the object is optional and will diff everything if left undefined? One thing to consider is what to do when the section is not defined in the diff object. I wouldn't mind a design where the object is used to exclude things rather than include them? I'm not sure what makes more sense as a default though.

File: src/diff.ts (hunk 1 above)
Comment (Julusian) [reply src/diff.ts:14]: Yeah I think so

**Comment by Julusian** [comment]
The object is not optional, but there is a helper `DiffAllObject()` to create the object to pass to diff everything. It could be made optional, but I like it being required to make it clearer in the api.

I think that being an object of what to include is better from a breaking-api perspective.
Whenever we support diffing new things, that is a breaking change as features which existed but were not controlled before are now controlled. This is why this PR which adds more sections has not been merged, because then it would break the nrk uses of this library.

But by having a list of inclusions, while it could still be classed as a breaking change to start diffing new features, we give an easy way to not opt-into that. By building your own inclusions object (which could be serialized to json or something), you will get the same behaviour on newer versions of the library.

**Review by baltedewit — APPROVED** [review]
This looks good to me on a principal level

**Pull request merged**